Add icon to deep link resource

JanHolger opened the pull request:
## Summary of Changes

The LtiDeepLinkResource was missing the url and thumbnail properties as described [inside the deep linking specification](https://www.imsglobal.org/spec/lti-dl/v2p0#deep-linking-response-example). This PR adds those properties and some more php-docs.

## Testing

I added tests for both the new class and the new properties

- [x] I have added automated tests for my changes
- [x] I ran `composer test` before opening this PR
- [x] I ran `composer lint-fix` before opening this PR

dbhynds commented on src/LtiDeepLinkResourceIcon.php:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @return $this
+     */
+    public function setWidth($width)
```
```suggestion
    public function setWidth(int $width)
```

dbhynds commented on src/LtiDeepLinkResourceIcon.php:
```diff
@@ -0,0 +1,90 @@
+    /**
+     * @return array
+     */
+    public function toArray()
```
```suggestion
    public function toArray(): array
```

dbhynds commented on tests/LtiDeepLinkResourceIconTest.php:
```diff
@@ -0,0 +1,90 @@
+    {
+        $result = $this->deepLinkResourceIcon->getUrl();
+
+        $this->assertEquals(null, $result);
```
```suggestion
        $this->assertNull($result);
```

dbhynds commented on tests/LtiDeepLinkResourceIconTest.php:
```diff
@@ -0,0 +1,90 @@
+    {
+        $result = $this->deepLinkResourceIcon->getWidth();
+
+        $this->assertEquals(null, $result);
```
```suggestion
        $this->assertNull($result);
```

dbhynds commented on tests/LtiDeepLinkResourceIconTest.php:
```diff
@@ -0,0 +1,90 @@
+    {
+        $result = $this->deepLinkResourceIcon->getHeight();
+
+        $this->assertEquals(null, $result);
```
```suggestion
        $this->assertNull($result);
```

dbhynds commented on tests/LtiDeepLinkResourceTest.php:
```diff
@@ -139,11 +172,26 @@ public function testItSetsTarget()
 
     public function testItCastsToArray()
     {
+        $icon = (new LtiDeepLinkResourceIcon())
```
Completely optional suggestion, but it looks cleaner when chained IMO:
```suggestion
        $icon = LtiDeepLinkResourceIcon::new()
```

dbhynds reviewed (changes requested):
Great job adhering to our existing design paradigms here. I have a bunch of recommended changes, but they're pretty minor overall. They basically fall into two categories:

1. We should aim for strict typing as much as possible.
2. `assertNull($result)` should be used instead of `assertEquals(null, $result)`

For the second point, `assertEquals()` does not perform a strict equality test. So `$this->assertEquals(null, false)` would pass since both values evaluate false without strict equality. Whereas `assertNull()` is a strict equality test. It's basically the difference between `null == false` vs `null === false`.

JanHolger commented on src/LtiDeepLinkResource.php:
```diff
@@ -9,98 +9,202 @@ class LtiDeepLinkResource
+     *
+     * @return $this
+     */
+    public function setLineitem($value)
```
I wasn't sure about whether to use strict typing because of backwards compatibility but i'm more than happy to see that you are using modern php (:

JanHolger commented:
I've replaced the php-doc comments with explicit typing.

I've also renamed $lineitem to $line_item and get-/setLineitem to get-/setLineItem to better fit the naming convention. As php isn't case sensitive this shouldn't break existing applications but as I'm not an expert in php please check that.

For DeepLinkResourceIcon I'm not quite sure about your design ideas with the `new` method but if I understood correctly this is only a helper for easier chaining so I've added all constructor parameters and passed them to the constructor.

dbhynds commented on src/LtiDeepLinkResourceIcon.php:
```diff
@@ -0,0 +1,67 @@
+    private $width;
+    private $height;
+
+    public function __construct(string $url, ?int $width = null, ?int $height = null)
```
Is `height` and `width` truly optional? In reading through the IMS docs, it doesn't indicate whether they are, but every example includes them. Have you tested not setting these values but only providing a URL? If so, what happened?

dbhynds replied:
if it _is_ required, I would also note that `public function setHeight(?int $height)` and `public function setWidth(?int $width)` should be changed to make those arguments required.

JanHolger replied:
Well I'm not really sure because the lti 1.3 docs are lacking a lot of information and it probably depends on how the platform handles the icons. The url was the only thing I was sure about because without a url the object doesn't make sense :smile: But as all examples in the IMS docs contain width and height I would agree that it's probably better to make them required.
